src/core/server/saved_objects/migrationsv2/types.ts

```diff
@@ -191,12 +191,25 @@ export type UpdateTargetMappingsWaitForTaskState = PostInitState & {
 export type OutdatedDocumentsSearch = PostInitState & {
   /** Search for outdated documents in the target index */
   readonly controlState: 'OUTDATED_DOCUMENTS_SEARCH';
+  readonly failedDocumentIds?: string[];
```

Comment:
might be simpler to not make this property optional otherwise we have three states:
 1. undefined
 2. empty array (no failures)
 3. non-empty array (some failures)

But we don't need or use (1), so we can rather initialize an empty array and then we don't need to use `failedDocumentIds!` in other places.